### DISCUSSION TITLE
Use hpcc-js-wasm with client-side React based on create-react-app

### DISCUSSION THREAD

**Discussion by andreash** [hpcc-systems/hpcc-js-wasm#103] [Q&A] [answered]
Hi,

thanks for this package - comes in handy for using graphviz in the browser.

I am using it with React and I have run into trouble to find a clean way to give the web-packed React-app a URL from whare it can load the .wasm file. As others, I ended up to **manually** copy the file into the public/js folder, from where it gets loaded.

I am wondering whether anybody might have an idea how to access the file from where it originally sits - in the node-modules hierarchy.

Best,
Andreas

**Comment by GordonSmith** [comment, marked as answer]
In general copying the wasm file to the same location the loader code has been bundled to is probably the simplest approach.  
To nominate a different location you can call `wasmFolder(...url...)` prior to making graphviz calls:
```

import { wasmFolder } from "@hpcc-js/wasm";

...

wasmFolder("../../node_modules/@hpcc-js/wasm/dist");

```
type of thing.

You may need to tweak the relative path depending on your root folder and the location of your bundled folder.

Note:  GitHub has introduced a new "Discussions" tab above ^^^^^ where these types of questions might be better located!

**Comment by GordonSmith** [comment]
Further - the reason I "prefer" the copying method is that I consider the wasm file to be a resource, similar to png, svg etc.  and if you don't copy it, you will be in danger of forgetting to do so when you migrate your bundled package to a production server.
